block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagRepresentationSyntax.scala

```diff
@@ -100,4 +104,44 @@ final class BlockDagRepresentationOps[F[_]: Sync](
       ib <- dag.invalidBlocks
       r  = ib.map(block => (block.blockHash, block.sender)).toMap
     } yield r
+
+  /** children that have block hash as the main parent */
+  def getSubordinateChildren(
+      mainParent: BlockHash
+  )(implicit bs: BlockStore[F]): F[List[BlockHash]] =
+    dag
+      .children(mainParent)
+      .flatMap(
+        _.traverse(
+          _.toList.filterA(
+            hash =>
+              BlockStore[F]
+                .getUnsafe(hash)
```

Comment:
We are waiting here in sequence for each block to be read from disk and decoded.